framework/lib/components/combo-picker/combo-picker.tsx

```diff
@@ -78,13 +91,20 @@ export const ComboPicker = ({
       <FormattedNumberInput
         width="100%"
         onChange={ handleInputChange }
-        defaultValue={ inputValue }
+        value={ inputValue }
         placeholder={ placeholder }
         size={ size }
         numberOfDecimals={ precision }
         preset={ formatPreset }
         disabled={ isDisabled }
         readOnly={ isReadOnly }
+        min={ min }
+        max={ max }
+        onBlur={ () => {
+          if (!is(Number, inputValue) && defaultToZeroIfEmpty) {
+            setInputValue(0)
+          }
```

Comment:
Check if inputValue isNil from ramda instead, it should be more direct?

Reply:
If the inputValue would be NaN for some reason, it would not be properly handled by isNil, so for all the checks on this component and MoneyInputField I'm just checking if it's a number instead, and if it's not a number than it's not valid